docs/adding-a-new-app.md

```diff
@@ -1,248 +1,178 @@
+# [*db_name*]
+#   The database name to use for the DATABASE_URL environment variable
 #
-class govuk::apps::my_app (
-  $port = 3456,
+class govuk::apps::myapp (
+  $port = 99999999,
+  $enabled = false,
+  $secret_key_base = undef,
+  $sentry_dsn = undef,
+  $oauth_id = undef,
+  $oauth_secret = undef,
+  $db_hostname = undef,
+  $db_password = undef,
+  $db_name = 'myapp_production',
 ) {
-  govuk::app { 'my-app':
+  $app_name = 'myapp'
+
+  $ensure = $enabled ? {
+    true  => 'present',
+    false => 'absent',
+  }
+
+  # see modules/govuk/manifests/app.pp for more options
+  govuk::app { $app_name:
+    ensure            => $ensure,
     app_type          => 'rack',
     port              => $port,
     sentry_dsn        => $sentry_dsn,
     vhost_ssl_only    => true,
-    health_check_path => '/',
+    health_check_path => '/healthcheck', # must return HTTP 200 for an unauthenticated request
+    deny_framing      => true,
+    asset_pipeline    => true,
   }
-}
-```
 
-The `health_check_path` must be an endpoint on the application returning a
-HTTP 200 status for an unauthenticated request. Monitoring checks for the application are
-configured using this path and fail if the request isn't successful.
+  govuk::app::envvar {
+    app               => $app_name,
+    ensure            => $ensure,
+    notify_service    => $enabled,
+  }
 
-If your app uses the default web rails web server (on the the GOV.UK stack, this is [Unicorn](https://rubygems.org/gems/unicorn/versions/5.1.0)), you will need to add an entry for `secret_key_base`.
+  govuk::app::envvar {
+    "${title}-SECRET_KEY_BASE":
+      varname        => 'SECRET_KEY_BASE',
+      value          => $secret_key_base;
+    "${title}-OAUTH_ID":
+      varname        => 'OAUTH_ID',
+      value          => $oauth_id;
+    "${title}-OAUTH_SECRET":
+      varname        => 'OAUTH_SECRET',
+      value          => $oauth_secret;
+  }
 
-```
-if $secret_key_base != undef {
-  govuk::app::envvar { "${title}-SECRET_KEY_BASE":
-  varname => 'SECRET_KEY_BASE',
-  value   => $secret_key_base,
+  if $::govuk_node_class !~ /^development$/ {
+    govuk::app::envvar::database_url { $app_name:
+      type           => 'postgresql',
+      username       => $app_name,
+      password       => $db_password,
+      host           => $db_hostname,
+      database       => $db_name,
+   }
+  }
 }
 ```
 
-The `secret_key_base` is used to encrypt user sessions. If it's not defined the application will not start. The value of
-`secret_key_base` should be encrypted in the `govuk-secrets` repo.
-
-Additional arguments can be specified to configure basic auth, monitoring checks and logging.
-These are defined and explained in `modules/govuk/manifests/app.pp`.
-
-## Database setup
-
-Create a file in `modules/govuk/manifests/apps/my_app` named `db.pp`:
+Also add a file in `modules/govuk/manifests/apps/my_app` named `db.pp`:
```

Comment:
I think since we've dropped the section on `Database setup` it's probably clearer if we say "If your application uses a Postgres/MySQL add a file to" (as I don't think we do this for MongoDB)